File: src/main/java/controller/BooksController.java
package controller;

import model.Book;
import spark.Request;
import spark.Response;
import spark.Route;

public class BooksController extends ModelController {

    public static Route index = (Request request, Response response) -> index(Book.class);
    public static Route show = (Request request, Response response) -> show(Book.class, request);
}

File: src/main/kotlin/controller/BooksController.kt
package controller

import model.Book
import service.GoodreadsService
import service.JsonService
import spark.Request
import spark.Response
import spark.Spark
import java.time.LocalDateTime

class BooksController : ModelController() {
    companion object {

        fun register(endpointUrl: String) {
            Spark.get(endpointUrl, index, JsonService.gson::toJson)
        }

        private var bookListUpdate: LocalDateTime? = null
        private var bookList: List<Book> = listOf()

        val index = fun(_: Request, _: Response): List<Book> {
            synchronized(this) {
                if (bookListUpdate?.isBefore(LocalDateTime.now().minusHours(1)) ?: true) {
                    bookList = GoodreadsService.getToReadList(37418936)
                    bookListUpdate = LocalDateTime.now()
                }
            }
            return bookList
        }
    }
}
